compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -1143,6 +1143,43 @@ LogicalResult InterfaceBindingSubspanOp::verify() {
     }
   }
 
+  if (auto memrefType = dyn_cast<MemRefType>(op.getType())) {
+    if (memrefType.getElementType().isIndex()) {
+      // Index type can have different bitwidths depending on the target
+      // environment.
+      return success();
+    }
+
+    SmallVector<int64_t> strides;
+    int64_t typeOffset;
+    if (failed(getStridesAndOffset(memrefType, strides, typeOffset))) {
+      // Stop verification if we cannot get static offset from the type.
+      return success();
+    }
+
+    if (ShapedType::isDynamic(typeOffset)) {
+      return success();
+    }
+
+    Value offset = getByteOffset();
+    APInt subspanOffset;
+    if (!offset) {
+      subspanOffset = APInt::getZero(64);
+    } else if (!matchPattern(offset, m_ConstantInt(&subspanOffset))) {
+      // Stop verification if we cannot get static offset from the op.
+      return success();
+    }
+
+    int32_t numBytes =
```

Comment:
This is not going to work. It has to be consistent with what is being done here : https://github.com/openxla/iree/pull/14647/files#diff-7ae3c63ccd638175a21c445e2cdba448dd11262ba74d1b88d73dd4a78668bed1R714

Reply:
Yeah was just about to point that out - getRoundedElementByteWidth is almost never going to work and we should actually remove it. You'd need getRoundedPhysicalStorageSize (which properly handles packing). But the "rounded" part of that is what will make this sketchy as it's mixing a type-erased byte bag produced from who knows where with typed-and-packed elements that don't line up.